arangod/Transaction/Manager.h

```diff
@@ -21,66 +21,87 @@
 /// @author Jan Steemann
 ////////////////////////////////////////////////////////////////////////////////
 
-#ifndef ARANGOD_MMFILES_MMFILES_TRANSACTION_MANAGER_H
-#define ARANGOD_MMFILES_MMFILES_TRANSACTION_MANAGER_H 1
+#ifndef ARANGOD_TRANSACTION_MANAGER_H
+#define ARANGOD_TRANSACTION_MANAGER_H 1
 
 #include "Basics/Common.h"
 #include "Basics/ReadWriteLock.h"
-#include "StorageEngine/TransactionManager.h"
+#include "Basics/ReadWriteSpinLock.h"
+#include "Basics/Result.h"
+#include "Transaction/Status.h"
+#include "VocBase/AccessMode.h"
 #include "VocBase/voc-types.h"
 
+#include <atomic>
+#include <vector>
+
 namespace arangodb {
+class TransactionState;
+// to be derived by storage engines
+struct TransactionData {};
+
+namespace transaction {
+class Context;
 
-class MMFilesTransactionManager final : public TransactionManager {
+class Manager final {
   static constexpr size_t numBuckets = 16;
+  static constexpr double defaultTTL = 10.0 * 60.0;   // 10 minutes
+  static constexpr double tombstoneTTL = 5.0 * 60.0;  // 5 minutes
 
  public:
-  MMFilesTransactionManager() : TransactionManager() {}
-  ~MMFilesTransactionManager() {}
+  Manager(bool keepData) : _keepTransactionData(keepData), _nrRunning(0) {}
```

Comment:
let's add `explicit` here.
```suggestion
  explicit Manager(bool keepData) : _keepTransactionData(keepData), _nrRunning(0) {}
```